eitprocessing/roi_selection/gridselection.py

```diff
@@ -0,0 +1,390 @@
+    regions. This behaviour is controlled by `split_rows` and `split_columns`.
+
+    If `split_rows` is `False` (default), rows will not be split between two groups. A warning will
+    be shown stating regions don't contain equal numbers of rows. The regions towards the top will
+    be larger. E.g., when a (5, 2) array is split in two vertical regions, the first region will
+    contain the first three rows, and the second region the last two rows.
+
+    If `split_rows` is `True`, e.g. a (5, 2) array that is split in two vertical regions, the first
+    region will contain the first two rows and half of each pixel of the third row. The second
+    region contains half of each pixel in the third row, and the last two rows.
+
+    `split_columns` has the same effect on columns as `split_rows` has on rows.
+
+    Regions are ordered according to C indexing order. The `matrix_layout()` method provides a map
+    showing how the regions are ordered.
+
+    Common grids are pre-defined:
+    - VentralAndDorsal: vertically divided into ventral and dorsal;
+    - RightAndLeft: horizontally divided into anatomical right and left; NB: anatomical right is
+    the left side of the matrix;
+    - FourLayers: vertically divided into ventral, mid-ventral, mid-dorsal and dorsal;
+    - Quadrants: vertically and horizontally divided into four quadrants.
+
+    Args:
+        v_split: The number of vertical regions. Must be 1 or larger.
+        h_split: The number of horizontal regions. Must be 1 or larger.
+        split_rows: Allows rows to be split over two regions.
+        split_columns: Allows columns to be split over two regions.
+
+    Examples:
+        >>> pixel_map = array([[ 1,  2,  3],
+                               [ 4,  5,  6],
+                               [ 7,  8,  9],
+                               [10, 11, 12],
+                               [13, 14, 15],
+                               [16, 17, 18]])
+        >>> gs = GridSelection(3, 1, split_pixels=False)
+        >>> matrices = gs.find_grid(pixel_map)
+        >>> matrices[0] * pixel_map
+        array([[1, 2, 3],
+               [4, 5, 6],
+               [0, 0, 0],
+               [0, 0, 0],
+               [0, 0, 0],
+               [0, 0, 0]])
+        >>> gs.matrix_layout()
+        array([[0],
+               [1],
+               [2]])
+        >>> gs2 = GridSelection(2, 2, split_pixels=True)
+        >>> matrices2 = gs.find_grid(pixel_map)
+        >>> gs2.matrix_layout()
+        array([[0, 1],
+               [2, 3]])
+        >>> matrices2[2]
+        array([[0. , 0. , 0. ],
+               [0. , 0. , 0. ],
+               [0. , 0. , 0. ],
+               [1. , 0.5, 0. ],
+               [1. , 0.5, 0. ],
+               [1. , 0.5, 0. ]])
+    """
+
+    v_split: int
+    h_split: int
+    split_rows: bool = False
+    split_columns: bool = False
+    ignore_nan_rows: bool = True
+    ignore_nan_columns: bool = True
```

Comment:
I think it would be nice to have a single argument for each of split and ignore, as I think that more often than not a user would want the same behavior for both.
We can set it so that it accepts either a bool or a 2-part tuple of bools (for horizontal and vertical).

```suggestion
    split_pixels: bool | tuple[bool, bool] = False
    ignore_nan_edges: bool | tuple[bool, bool] = True
```

Also, what would happen and what do we want to happen if the center row/column of an array is nan?